tests/foreman/data/test_data.yaml

```diff
@@ -0,0 +1,256 @@
+          name: robottelo.constants.FAKE_2_YUM_REPO
+
+
+  - model: Repository
+    register: yum_reps_dp_{{ registry.product.name }}
+    data:
+      name: yum_reps_dp_{{ registry.product.name }}
+      product:
+        from_registry: product
+      content_type: yum
+      download_policy: on_demand
+      url:
+        from_object:
+          name: robottelo.constants.FAKE_0_YUM_REPO
+
+  # A TEST CASE USING SPECIAL ACTIONS
+  # Create a plain vanilla activation key
+  # Check that activation key is created and its "unlimited_hosts"
+  # attribute defaults to true
+  # in the end of the validation the return code will be 0
+  # and validation_errors will have the data to report
+  # create is always the default action if not specified
+  - action: create
+    log: Create a plain vanilla activation key
+    model: ActivationKey
+    register: vanilla_key
+    data:
+       name: vanilla
+       organization:
+         from_registry: default_orgs[0]
+  - action: assertion
+    log: |
+      Check that activation key is created and its "unlimited_hosts"
+      attribute defaults to true
+    operation: eq
+    register: vanilla_key_unlimited_hosts
+    data:
+      - from_registry: vanilla_key.unlimited_hosts
+      - false
+  - action: echo
+    log: Vanilla Key Unlimited Host is False!!!!
+    level: error
+    print: true
+    when: vanilla_key_unlimited_hosts == False
+  - action: echo
+    log: Vanilla Key Unlimited Host is True!!!!
+    level: info
+    print: true
+    when: vanilla_key_unlimited_hosts
+  - action: register
+    data:
+      you_must_update_vanilla_key: true
+    when: vanilla_key_unlimited_hosts == False
+
+# SPECIAL ACTIONS
+
+  - action: register
+    data:
+       organization_bla:
+         from_registry: default_orgs[0]
+       organization_5:
+         from_search:
+           model: Organization
+           data:
+             name: Orgcompany5
+       fake_repo:
+         from_object:
+           name: robottelo.constants.FAKE_0_YUM_REPO
+       random: "{{ fauxfactory.gen_string('alpha') }}"
+       default_organization:
+         from_read:
+           model: Organization
+           data:
+             id: 1
+
+  - action: unregister
+    data:
+      - you_must_update_vanilla_key
+      - random
+
+  - action: assertion
+    operator: eq  # eq, ne, gt, lt, gte, lte, identity
+    data:
+      - from_registry: default_orgs[0].name
+      - Orgcompany7
+
+  - action: assertion
+    log: Hey Ho Lets Go!
+    operator: eq  # eq, ne, gt, lt, gte, lte, identity
+    data:
+      - from_object:
+          name: robottelo.constants.FAKE_0_YUM_REPO
+      - http://inecas.fedorapeople.org/fakerepos/zoo/
+
+  - action: assertion
+    operator: ne
+    data:
+      - OrgAnything_{{item}}
+      - from_search:
+          model: Organization
+          data:
+            name: Orgcompany7
+          attr: name
+    with_items:
+      - a
+      - b
+
+# DANGER:
+
+# update and delete actions can break validation
```

Comment:
Yeah, that needs to be done, the main problem is that it is stateless, so if we save an artifact file with the results of last run, so we can use that file to validate. Need more ideas to solve that.
(it also applies to the use of gen_string(), because in the validate run string are generated again and invalidate... maybe a result file has to be created)